### PR TITLE
defaulted hidden state for added complex fields in collection to false

### PR DESCRIPTION
**Pull request by phillip-whitaker**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-3983

#### The Issue
When a collection that contains complex fields is added as part of a collection the fields with show condition are momentarily displayed before the `conditional-show-form.directive.ts` kicks in and hides it. This is because the subscription has a debounce(100) on it. The debounce is there to help performance.

#### The Fix
As I saw it, there are two fixes to this bug, remove the debounce or default added collection complex fields to hidden = true for fields that have show_condition logic. I chose the latter because I think is more sustainable and I know that the show/hide logic all being changed soon too so this fix will withstand the changes to the current directive.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x-x-x] No
```

### CODE REVIEW

**Review by Mo-Lala84 — APPROVED** [review]
lgtm

File: src/shared/components/palette/collection/write-collection-field.component.ts [hunk 1]
```diff
@@ -156,11 +156,19 @@ export class WriteCollectionFieldComponent extends AbstractFieldWriteComponent i
 
   private newCaseField(id: string, item, index, isNew = false) {
     const isNotAuthorisedToUpdate = !isNew && this.isNotAuthorisedToUpdate(index);
+
+    const fieldType = plainToClassFromExist(new FieldType(), this.caseField.field_type.collection_field_type);
+    if (fieldType.complex_fields) {
+      fieldType.complex_fields
+        .filter((cf: CaseField) => cf.show_condition !== null)
```
Comment (paul-graham) [inline src/shared/components/palette/collection/write-collection-field.component.ts:163]: Do we know for sure this is always `null` when it's not set? It's never `undefined` or `''`? `!!cf.show_condition` would account for all of those, which might be a better option.

**Review by paul-graham — COMMENTED** [review]
Just a question around the filtering mechanism...

File: src/shared/components/palette/collection/write-collection-field.component.ts [hunk 2]
```diff
@@ -156,11 +156,19 @@ export class WriteCollectionFieldComponent extends AbstractFieldWriteComponent i
 
   private newCaseField(id: string, item, index, isNew = false) {
     const isNotAuthorisedToUpdate = !isNew && this.isNotAuthorisedToUpdate(index);
+
+    const fieldType = plainToClassFromExist(new FieldType(), this.caseField.field_type.collection_field_type);
+    if (fieldType.complex_fields) {
+      fieldType.complex_fields
+        .filter((cf: CaseField) => !cf.show_condition)
```
Comment (paul-graham) [inline src/shared/components/palette/collection/write-collection-field.component.ts:163]: This should be a double bang, not a single. You're filtering out anything _with_ a `show_condition` now! 😂

**Review by paul-graham — CHANGES_REQUESTED** [review]

**Review by paul-graham — APPROVED** [review]
Sorted. Thanks for fixing the double bang, @phillip-whitaker.

**Review by paul-graham — DISMISSED** [review]

**Review by paul-graham — APPROVED** [review]

**Pull request merged**